From: Marco Castelluccio <mcastelluccio@mozilla.com>
Date: Wed, 12 Apr 2023 18:34:41 +0200
Subject: [PATCH] Bump quick-xml from 0.26.0 to 0.28.1 (#999)

Bumps [quick-xml](https://github.com/tafia/quick-xml) from 0.26.0 to 0.28.1.
- [Release notes](https://github.com/tafia/quick-xml/releases)
- [Changelog](https://github.com/tafia/quick-xml/blob/master/Changelog.md)
- [Commits](tafia/quick-xml@v0.26.0...v0.28.1)

---
updated-dependencies:
- dependency-name: quick-xml
  dependency-type: direct:production
  update-type: version-update:semver-minor
---
 Cargo.toml    | 2 +-
 src/parser.rs | 3 ++-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/Cargo.toml b/Cargo.toml
index dce1fb101..c4cb36f37 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -50,7 +50,7 @@ log = "0.4"
 version = "1.13"
 
 [dependencies.quick-xml]
-version = "0.26"
+version = ">= 0.27"
 
 [dependencies.rayon]
 version = "^1"
diff --git a/src/parser.rs b/src/parser.rs
index 991ff11bc..761084577 100644
--- a/src/parser.rs
+++ b/src/parser.rs
@@ -8,6 +8,7 @@ use std::io::{self, BufRead, BufReader, Read};
 use std::num::ParseIntError;
 use std::path::Path;
 use std::str;
+use std::sync::Arc;
 
 use log::error;
 
@@ -37,7 +38,7 @@ impl From<io::Error> for ParserError {
 impl From<quick_xml::Error> for ParserError {
     fn from(err: quick_xml::Error) -> ParserError {
         match err {
-            quick_xml::Error::Io(e) => ParserError::Io(e),
+            quick_xml::Error::Io(e) => ParserError::Io(Arc::try_unwrap(e).unwrap()),
             _ => ParserError::Parse(format!("{:?}", err)),
         }
     }
